tests/std/tests/P0896R4_ranges_alg_find_first_of/test.cpp

```diff
@@ -2,92 +2,86 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <algorithm>
-#include <array>
 #include <cassert>
 #include <concepts>
 #include <ranges>
 #include <utility>
 
 #include <range_algorithm_support.hpp>
+using namespace std;
+using P = std::pair<int, int>;
 
-constexpr void smoke_test() {
-    using ranges::find_first_of, ranges::iterator_t, std::array, std::same_as, std::to_address;
-    using P = std::pair<int, int>;
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::find_first_of(borrowed<false>{}, borrowed<true>{})), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::find_first_of(borrowed<true>{}, borrowed<true>{})), int*>);
 
-    // Validate dangling story
-    STATIC_ASSERT(same_as<decltype(find_first_of(borrowed<false>{}, array<int, 42>{})), ranges::dangling>);
-    STATIC_ASSERT(same_as<decltype(find_first_of(borrowed<true>{}, array<int, 42>{})), int*>);
-
-    const array pairs = {P{0, 42}, P{1, 42}, P{2, 42}, P{3, 42}, P{4, 42}, P{5, 42}, P{6, 42}};
-
-    const auto pred = [](const int x, const int y) { return x == y + 1; };
+struct instantiator {
+    static constexpr P haystack[7]      = {{0, 42}, {1, 42}, {2, 42}, {3, 42}, {4, 42}, {5, 42}, {6, 42}};
+    static constexpr int good_needle[2] = {29, 1};
+    static constexpr int bad_needle[2]  = {29, 17};
 
-    const array good_needle = {29, 1};
-    {
-        // Validate range overload [found case]
-        auto result = find_first_of(basic_borrowed_range{pairs}, good_needle, pred, get_first);
-        STATIC_ASSERT(same_as<decltype(result), iterator_t<basic_borrowed_range<const P>>>);
-        assert(result.peek() == pairs.data() + 2);
-    }
-    {
-        // Validate iterator + sentinel overload [found case]
-        basic_borrowed_range wrapped_pairs{pairs};
-        auto result = find_first_of(
-            wrapped_pairs.begin(), wrapped_pairs.end(), good_needle.begin(), good_needle.end(), pred, get_first);
-        STATIC_ASSERT(same_as<decltype(result), iterator_t<basic_borrowed_range<const P>>>);
-        assert(result.peek() == pairs.data() + 2);
-    }
+    template <ranges::input_range Read1, ranges::forward_range Read2>
+    static constexpr void call() {
+        using ranges::find_first_of, ranges::iterator_t;
+        const auto pred = [](const int x, const int y) { return x == y + 1; };

```

Comment:
Putting this at class or namespace scope so we have one lambda type per program instead of one lambda type per combination of `(Read1, Read2)` will probably save a lot of compile time that's spent instantiating identical templates.

Reply:
That is correct, moved out